multinet/api/tests/test_network.py

```diff
@@ -417,9 +417,11 @@ def test_network_rest_retrieve_tables_all(
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('type', ['node', 'edge'])
+@pytest.mark.parametrize(
+    'type,success', [('node', True), ('edge', True), ('all', True), ('foo', False)]
+)
 def test_network_rest_retrieve_tables_type(
-    workspace: Workspace, user: User, authenticated_api_client: APIClient, type: str
+    workspace: Workspace, user: User, authenticated_api_client: APIClient, type: str, success: bool
```

Comment:
`str` doesn't seem like the right type for `type`; it should be a `Union` of the literals that make up the valid set of types.

This seems nitpicky, I know, but the point of using types is to squeeze as much correctness out of the code at the typechecking phase. Unions of string literals to express enum-typed values is a very powerful feature.

Reply:
I think string might be more appropriate here, since the test cases here don't necessarily pass in a "correct" value for `type`. Let me know if there's a subtlety here that I'm not quite picking up on though.

Reply:
You're right, we're explicitly testing for `type == 'foo'` in these test cases.